pyiron_base/master/parallel.py

```diff
@@ -940,7 +940,7 @@ def job_name(self, parameter):
         Returns:
             str: job name for the next child job
         """
-        return self._master.ref_job.job_name + "_" + str(self._childcounter)
+        return '_'.join([self._master.job_name, self._master.ref_job.job_name, str(self._childcounter)])
```

Comment:
On the implementation side, if we adopt this, it should rather be changed below, in `.next:964`, otherwise sub classes of `JobGenerator` need to reimplement that logic.